CMakeLists.txt

```diff
@@ -1,17 +1,83 @@
 cmake_minimum_required(VERSION 3.10)
 
-project(cxxeffect)
+project(cxxeffect CXX)
 
-# add the executable
-add_executable(cxxeffect
-    src/stream.hpp
-    src/stream.cpp
-    src/task.hpp
-    src/task.cpp
-    src/eff.hpp
-    src/eff.cpp)
+# We want to export compile commands b/c it's useful for ccls
+set(CMAKE_EXPORT_COMPILE_COMMANDS true)
 
-set_property(TARGET cxxeffect PROPERTY CXX_STANDARD 17)
```

Comment:
Alright, so just adding back the property for CXX20 should be good here. I'm not a CMake expert, would that negate the need for the `-std`/`/std` compiler flags? I feel like it should.

Reply:
Probably! I'll test it!

Reply:
Update: We still need the -std / std compile flags. CMake doesn't add `-std=c++20` even when we have `set_property(TARGET cxxeffect PROPERTY CXX_STANDARD 20)`

Reply:
This seems relevant: https://stackoverflow.com/a/20165519/4580217

Reply:
CMake doesn't have a feature flag for coroutines, and `target_compile_features(cxxeffect PRIVATE cxx_std_20)` should be automatically added when `CXX_STANDARD` is set to 20...

Reply:
Ah, what happens when you set `CXX_STANDARD_REQUIRED` instead of `CXX_STANDARD`?

Reply:
That still doesn't do it, but if we require C++20 as part of the interface of cxxeffect it works! i will update CMakeLists.txt.
```cmake
target_compile_features(cxxeffect INTERFACE cxx_std_20)
```

Reply:
Alright, we should do both `CXX_STANDARD_REQUIRED` and `cxx_std_20` then 👍 

Really odd that this happens... I feel like CMake should be doing the latter from the former.